src/main/java/frc/robot/subsystems/MecanumDriveSubsystem.java

```diff
@@ -43,14 +54,55 @@ public MecanumDriveSubsystem() {
     rearRight.burnFlash();
 
     driveBase = new MecanumDrive(frontLeft, rearLeft, frontRight, rearRight);
+
+    if (RobotBase.isSimulation()) {
+      REVPhysicsSim.getInstance().addSparkMax(frontLeft, DCMotor.getNEO(1));
+    REVPhysicsSim.getInstance().addSparkMax(frontRight, DCMotor.getNEO(1));
+    REVPhysicsSim.getInstance().addSparkMax(rearLeft, DCMotor.getNEO(1));
+    REVPhysicsSim.getInstance().addSparkMax(rearRight, DCMotor.getNEO(1));
+
+    }
+
+     x=0;
+    
+  }
+
+  public Map<String,RelativeEncoder> GetEncoders()
+  {
+    Map<String, RelativeEncoder> Encoders = new HashMap<String,RelativeEncoder>();
+    Encoders.put("Front Left", frontLeft.getEncoder());
+    Encoders.put("Front Right", frontRight.getEncoder());
+    Encoders.put("Rear Left", rearLeft.getEncoder());
+    Encoders.put("Rear Right", rearRight.getEncoder());
+
+    return Encoders;
   }
 
   public void drive(double forwardSpeed, double rightSpeed, double rotatinalSpeed) {
     driveBase.driveCartesian(forwardSpeed, rightSpeed, rotatinalSpeed);
+
+    SmartDashboard.putNumber("y", y);
+    y++;
   }
 
   @Override
   public void periodic() {
-    // This method will be called once per scheduler run
+  // This method will be called once per scheduler run
+    // double frontLeftSpeed = frontLeft.getEncoder().getVelocity();
```

Comment:
Delete commented out code, or use it.  It wouldn't be bad for the periodic function to post drive speeds to SmartDashboard.

Reply:
That I left in there while I tested the periodic code so I could figure out where the problem I was seeing was coming from.